### PR TITLE
Update UI to follow GUI Figma prototype

### PR DESCRIPTION
**Pull request by yl-ang**
- In the bottom half of LibTask, create 2 panels for patron list and book list.
- In each panel contains the patron cards and book cards respectively.
- Each card consists of relevant details that closely follow the team's shared document.
- Checks for text truncation in place.
-  ~~Proper checks for book status to render the relevant tags.~~

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -305,6 +305,8 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `*`      | Impatient user                                            | be able to experience reasonable response time while using the application    | I can do my job more efficiently                                                                                                      |
 | `*`      | Easily overwhelmed request handler                        | be able to prioritize my book requests                                        | I can effectively handle requests                                                                                                     |
+| `***`    | Librarian                                                 | be able to view patron and book details through a GUI                         |                                                                                                                                       |
+| `***`    | Librarian                                                 | have visual feedback of the commands I executed                               | I will know if my commands have been successfully executed                                                                            |
```
Comment (Halpfrog) [inline docs/DeveloperGuide.md:309]: Maybe can move these to the top cause it has been assigned a higher priority.

File: src/main/java/seedu/address/ui/BookCard.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    @FXML
+    private Label authors;
+    @FXML
+    private Label isbn;
+    @FXML
+    private FlowPane bookCategoryTags;
+    @FXML
+    private FlowPane bookAvailableTag;
+    @FXML
+    private FlowPane bookBorrowTag;
```
Comment (bingsen0806) [inline src/main/java/seedu/address/ui/BookCard.java:46]: Is it necessary to have two FlowPane since they are mutually exclusive? Only one will be shown at a time. How do you enforce that no two FlowPane will be shown at any given time?

Comment (bingsen0806) [inline src/main/java/seedu/address/ui/BookCard.java:46]: Is it necessary to have two FlowPane since they are mutually exclusive? Only one will be shown at a time. How do you enforce that no two FlowPane will be shown at any given time?

File: src/main/java/seedu/address/ui/BookCard.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        // Author check needed since author is an optional parameter in book command
+        if (!book.getAuthors().isEmpty()) {
+            authors.setVisible(true);
+            authors.setManaged(true);
+            String authorsString = book.getAuthors().stream()
+                    .sorted(Comparator.comparing(author -> author.toString()))
+                    .map(Author::toString)
+                    .collect(Collectors.joining(", "));
+            authors.setText("Author: " + authorsString);
+        }
```
Comment (bingsen0806) [inline src/main/java/seedu/address/ui/BookCard.java:66]: Good use of streams! Nice checking of empty case

File: src/main/java/seedu/address/ui/BookListPanel.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.book.Book;
+
+/**
+ * Panel containing the list of persons.
```
Comment (bingsen0806) [inline src/main/java/seedu/address/ui/BookListPanel.java:14]: Shouldn't this be list of books?

File: src/main/resources/view/DarkTheme.css [hunk 5]
```diff
@@ -337,16 +349,59 @@
     -fx-background-radius: 0;
 }
 
-#tags {
+/** Different style classes for different types of tags **/
```
Comment (bingsen0806) [inline src/main/resources/view/DarkTheme.css:352]: Nice consideration!

File: src/test/data/JsonSerializableAddressBookTest/lengthyBooksAddressBook.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        "SciFi", "Thrillers", "Drama", "Poetry", "Media", "Nonfiction", "Horror1", "Literary1", "Romance1", "Crime1",
+        "SciFi1", "Thrillers1", "Drama1", "Poetry1", "Media1" , "Media2", "Media3", "Media4", "Media5", "Media6",
+        "Media7", "Media8", "Media9", "Media10"],
+      "authors" : [ "accusamus acutum aegritudine alios aliquip allevatio antiqua arbitramur arbitrium arcu augeri captet censes cetero cogitavisse comparare conscientiam contentus cupiditatum deditum democritus depravata deserunt desiderant desistunt detracta didicerimus dirigentur displicet dissensio disseretur distinguique eget elit eo epicurei equos exercitus exorsus fastidium faucibus felis foedus fuissent generis geometrica gravida illam impetus incursione indignae inhaererent initia inquam inter ipsas ipsi iriure lacinia licet mel mi minim mors muniti netus noster nostro occultarum odioque omnino oratoribus oriantur ornateque percipit plura pluribus possint praetulerit privamur probamus procedat propemodum pulchraeque quaerendi quanti quisque rudem sensum sole stultorum successerit successionem suspicor tranquilli unde varias videre voce vulgo "]
```
Comment (bingsen0806) [inline src/test/data/JsonSerializableAddressBookTest/lengthyBooksAddressBook.json:17]: Nice edge case testing!

**Review by bingsen0806 — APPROVED** [review]
LGTM! Good to see that you tested the UI extensively and created stubs even before the actual class is made available.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (yl-ang) [reply docs/DeveloperGuide.md:309]: Point taken, I think our user stories will need to rerank them after everyone added a priority for their assigned user stories. Then, we can create a branch to update UG.

File: src/main/java/seedu/address/ui/BookCard.java (hunk 3 above)
Comment (yl-ang) [reply src/main/java/seedu/address/ui/BookCard.java:66]: Thank you :D

File: src/main/java/seedu/address/ui/BookListPanel.java (hunk 4 above)
Comment (yl-ang) [reply src/main/java/seedu/address/ui/BookListPanel.java:14]: Thank you, I will update the doc to list of books!

File: src/test/data/JsonSerializableAddressBookTest/lengthyBooksAddressBook.json (hunk 6 above)
Comment (yl-ang) [reply src/test/data/JsonSerializableAddressBookTest/lengthyBooksAddressBook.json:17]: Thank you, will update more edge cases to ensure that our program doesn't break.

File: src/main/java/seedu/address/ui/BookCard.java (hunk 2 above)
Comment (yl-ang) [reply src/main/java/seedu/address/ui/BookCard.java:46]: By default, if there are no tags for that FlowPlane, it won't be rendered in the GUI. So we don't need to worry if the "Borrowed" and "Available" will be shown at the same time.

The control flow below will make sure that only one FlowPlane will be shown for each book.

```java
if (book.isAvailable()) {
    bookAvailableTag.getChildren().add(new Label("Available"));
} else {
    Collections.addAll(bookBorrowTag.getChildren(), new Label("Borrowed"),
            new Label(book.getDate()));
}
```

image
